extensions/common/metadata_object.cc

```diff
@@ -0,0 +1,227 @@
+      break;
+    case Istio::Common::WorkloadType::KUBERNETES_POD:
+      owner = fbb.CreateString(absl::StrCat(owner_prefix, obj.namespace_name_,
+                                            "/pods/", obj.workload_name_));
+      break;
+  }
+  labels.push_back(Wasm::Common::CreateKeyVal(
+      fbb, fbb.CreateString("service.istio.io/canonical-name"),
+      fbb.CreateString(toStdStringView(obj.canonical_name_))));
+  labels.push_back(Wasm::Common::CreateKeyVal(
+      fbb, fbb.CreateString("service.istio.io/canonical-revision"),
+      fbb.CreateString(toStdStringView(obj.canonical_revision_))));
+  labels.push_back(Wasm::Common::CreateKeyVal(
+      fbb, fbb.CreateString("app"),
+      fbb.CreateString(toStdStringView(obj.app_name_))));
+  labels.push_back(Wasm::Common::CreateKeyVal(
+      fbb, fbb.CreateString("version"),
+      fbb.CreateString(toStdStringView(obj.app_version_))));
+  // TODO: containers, ips, mesh id ?
+  auto labels_offset = fbb.CreateVectorOfSortedTables(&labels);
+  Wasm::Common::FlatNodeBuilder node(fbb);
+  node.add_name(name);
+  node.add_cluster_id(cluster);
+  node.add_namespace_(namespace_);
+  node.add_workload_name(workload_name);
+  node.add_owner(owner);
+  node.add_labels(labels_offset);
+  auto data = node.Finish();
+  fbb.Finish(data);
+  return fbb.Release();
+}
+
+Istio::Common::WorkloadMetadataObject convertFlatNodeToWorkloadMetadata(
+    const Wasm::Common::FlatNode& node) {
+  const absl::string_view instance = toAbslStringView(node.name());
+  const absl::string_view cluster = toAbslStringView(node.cluster_id());
+  const absl::string_view workload = toAbslStringView(node.workload_name());
+  const absl::string_view namespace_name = toAbslStringView(node.namespace_());
+  const auto* labels = node.labels();
+
+  const auto* name_iter =
+      labels->LookupByKey("service.istio.io/canonical-name");
+  const auto* name = name_iter ? name_iter->value() : nullptr;
+  const absl::string_view canonical_name = toAbslStringView(name);
+
+  const auto* revision_iter =
+      labels->LookupByKey("service.istio.io/canonical-revision");
+  const auto* revision = revision_iter ? revision_iter->value() : nullptr;
+  const absl::string_view canonical_revision = toAbslStringView(revision);
+
+  const auto* app_iter = labels->LookupByKey("app");
+  const auto* app = app_iter ? app_iter->value() : nullptr;
+  const absl::string_view app_name = toAbslStringView(app);
+
+  const auto* version_iter = labels->LookupByKey("version");
+  const auto* version = version_iter ? version_iter->value() : nullptr;
+  const absl::string_view app_version = toAbslStringView(version);
+
+  Istio::Common::WorkloadType workload_type =
+      Istio::Common::WorkloadType::KUBERNETES_POD;
+  // Strip "/workload_name" and check for workload type.
+  absl::string_view owner = toAbslStringView(node.owner());
+  owner.remove_suffix(workload.size() + 1);
+  if (absl::EndsWith(owner, "/deployments")) {
```

Comment:
auto owners = absl.StrSplit(owner, "/")];
switch (owners[len-1]){
case 'deployments':
case 'cornjobs':
}

Reply:
This would be doing more work since it's a long string. Probably easier to find last index and just tokenize/switch on the last part.

Reply:
avoid multi `absl:EndsWith`

Reply:
Yeah, will fix this with switch. Thanks.

Reply:
Done.